src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -137,73 +114,83 @@ export const SEQUENTIAL_COLOR_SCHEMES = ['blues', 'greens', 'greys', 'purples',
-    } else if (nestedProp === 'scheme') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: isContinuous(fieldDef) ? SEQUENTIAL_COLOR_SCHEMES : CATEGORICAL_COLOR_SCHEMES,
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    } else if (nestedProp === 'range' || nestedProp === 'domain') {
-      return generateSchema(prop, nestedProp, isDiscrete(fieldDef) ?
-        DEFAULT_DISCRETE_SCALE_ARRAY_UISCHEMA : DEFAULT_CONTINUOUS_SCALE_ARRAY_UISCHEMA, propTab, baseUISchema);
-    }
-  } else if (prop === 'axis') {
-    if (nestedProp === 'orient') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: channel === 'y' ? ['left', 'right'] : ['top', 'bottom']
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    } else if (nestedProp === 'title') {
-      return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-    }
-  } else if (prop === 'stack') {
-    const propertySchema = {
-      ...baseUISchema,
-      enum: (vlSchema as any).definitions.StackOffset.enum,
-    };
-    return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-  } else if (prop === 'size') {
-    return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-  } else if (prop === 'legend') {
-    if (nestedProp === 'orient') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: (vlSchema as any).definitions.LegendOrient.enum
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    } else if (nestedProp === 'title') {
-      return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-    } else if (nestedProp === 'type') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: (vlSchema as any).definitions.Legend.properties.type.enum,
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    }
-  } else if (prop === 'format') {
-    return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-  } else {
-    throw new Error('Property combination not recognized');
+      return generateScaleEditorSchema(nestedProp, scaleTypes, fieldDef, baseUISchema);
+    case 'axis':
+      return generateAxisEditorSchema(nestedProp, channel as Channel, baseUISchema, propertyKey);
+    case 'stack':
+      return generateStackEditorSchema(baseUISchema);
+    case 'legend':
+      return generateLegendEditorSchema(nestedProp, baseUISchema, propertyKey);
+    case 'format':
+      return generateFormatEditorSchema(baseUISchema);
+    default:
+      throw new Error('Property combination not recognized');
+  }
+}
+
+function generateFormatEditorSchema(baseUISchema: SchemaProperty) {
```

Comment:
For non-nested ones, these are one-liners anyway -- you can just return above -- avoid unnecessary code jump.  

(For nested ones, code jump is more justified as switch within switch is very hard to read.)